### PR TITLE
Update Example Dependencies

### PR DESCRIPTION
**Pull request by tommy-waltmann**
## Description

This PR updates examples using old versions of hoomd, gsd, and matplotlib.

## Checklist:

- [ ] I agree with the terms of the [**freud Contributor Agreement**](https://github.com/glotzerlab/freud-examples/blob/master/ContributorAgreement.md).
- [ ] My code follows the code style of this project.
- [ ] I have updated the [index notebook](https://github.com/glotzerlab/freud-examples/blob/master/index.ipynb) if needed.
- [ ] My name is on the [contributors list](https://github.com/glotzerlab/freud-examples/blob/master/contributors.txt).

### CODE REVIEW

File: examples/Benchmarking RDF against MDAnalysis.ipynb [hunk 1]
```diff
@@ -12,7 +12,17 @@
    "cell_type": "code",
```
Comment (tommy-waltmann) [inline examples/Benchmarking RDF against MDAnalysis.ipynb:12]: <p>I'm not sure where this warning comes from, but I'd like to remove it before merging this PR. It appears in a few notebooks.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/glotzerlab/freud-examples/pull/76/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

File: examples/HOOMD-MC-W6/HOOMD-MC-W6.ipynb [hunk 2]
```diff
@@ -19,10 +19,23 @@
    "cell_type": "code",
```
Comment (tommy-waltmann) [inline examples/HOOMD-MC-W6/HOOMD-MC-W6.ipynb:19]: <p>here too.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/glotzerlab/freud-examples/pull/76/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

File: examples/Visualization with plato.ipynb [hunk 3]
```diff
@@ -14,53 +14,70 @@
    "metadata": {},
```
Comment (tommy-waltmann) [inline examples/Visualization with plato.ipynb:14]: <p>and here</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/glotzerlab/freud-examples/pull/76/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

File: examples/Benchmarking RDF against MDAnalysis.ipynb (hunk 1 above)
Comment (joaander) [reply examples/Benchmarking RDF against MDAnalysis.ipynb:12]: This looks like a broken numpy installation.

**Comment by tommy-waltmann** [comment]
This PR will now resolve #71 #72 #74 too

**Review by DomFijan — APPROVED** [review]

**Comment by DomFijan** [comment]
One of the PMFT examples is failing in the CI, due to missing input file?

**Comment by tommy-waltmann** [comment]
> One of the PMFT examples is failing in the CI, due to missing input file?

I have never been able to reproduce the failure locally; the file is there. For some reason (only on py39), it thinks it can't find the file.

**Pull request merged**